src/type_checker/checker.rs

```diff
@@ -378,7 +369,7 @@ impl<B: Backend> TypeChecker<B> {
                 // check that expression is a bigint
                 let idx_typ = self.compute_type(idx, typed_fn_env)?;
                 match idx_typ.map(|t| t.typ) {
-                    Some(TyKind::BigInt) => (),
+                    Some(TyKind::Field { constant: true }) => (),
```

Comment:
btw probably you should have targeted the new const generic branch :D as you will have to add more changes for default array init also

Reply:
agreed, I have just change this PR's base branch to the const generic branch.